harvester_e2e_tests/integrations/test_4_vm_template.py

```diff
@@ -0,0 +1,195 @@
+                assert 200 == code, (code, data)
+                ns, name = data['spec']['defaultVersionId'].split('/')
+            except (AssertionError, ValueError):
+                # ValueError: version is not created yet, so `defaultVersionId` will be empty
+                pass
+            else:
+                code, data = api_client.templates.get_version(name, ns)
+                conds = data.get('status', {}).get('conditions', [])
+                if conds and all('True' == c['status'] for c in conds):
+                    tmpl_spec = api_client.templates.Spec.from_dict(data)
+                    break
+            sleep(5)
+        else:
+            raise AssertionError(
+                "Failed to create template with status:\n"
+                f"{data.get('status')}\n"
+                f"API Status({code}): {data}"
+            )
+
+        # Create new VM from the template
+        tmpl_vm_name = f"tmpl-{unique_name}"
+        code, data = api_client.vms.create(tmpl_vm_name, tmpl_spec)
+        assert 201 == code, (code, data)
+
+        vm_got_ips, (code, data) = vm_checker.wait_ip_addresses(tmpl_vm_name, ["default"])
+        assert vm_got_ips, (
+            f"Failed to Start VM({tmpl_vm_name}) with errors:\n"
+            f"Status: {data.get('status')}\n"
+            f"API Status({code}): {data}"
+        )
+
+        # Login to VM and check the data is consistent
+        vm_ip = next(iface['ipAddress'] for iface in data['status']['interfaces']
+                     if iface['name'] == 'default')
+        code, data = api_client.hosts.get(data['status']['nodeName'])
+        host_ip = next(addr['address'] for addr in data['status']['addresses']
+                       if addr['type'] == 'InternalIP')
+        with vm_shell_from_host(host_ip, vm_ip, ssh_user, pkey=pri_key) as sh:
+            cloud_inited, (out, err) = vm_checker.wait_cloudinit_done(sh)
+            assert cloud_inited, (
+                f"VM {tmpl_vm_name} Started {vm_checker.wait_timeout} seconds"
+                f", but cloud-init still in {out}"
+            )
+            out, err = sh.exec_command("md5sum -c ./generate_file.md5")
+            assert not err, (out, err)
+            vm2_md5, err = sh.exec_command("cat ./generate_file.md5")
+            assert not err, (vm2_md5, err)
+            assert vm1_md5 == vm2_md5
+
+        # teardown
+        api_client.vms.delete(tmpl_vm_name)
+        endtime = datetime.now() + timedelta(seconds=wait_timeout)
+        while endtime > datetime.now():
+            code, data = api_client.vms.get_status(tmpl_vm_name)
+            if 404 == code:
+                break
+            sleep(3)
+
+        for vol in tmpl_spec.volumes:
+            vol_name = vol['volume']['persistentVolumeClaim']['claimName']
+            api_client.volumes.delete(vol_name)
+
+        code, data = api_client.templates.delete(unique_name)
+        assert 200 == code, (code, data)
```

Comment:
Please confirm this DELETE do response `200`
https://github.com/harvester/tests/blob/97cbba4202e7b233bde6e277ea5d962af3de630a/apiclient/harvester_api/managers/templates.py#L8

Although UI uses different `v1/harvester/harvesterhci.io.virtualmachinetemplates/default/test` but it replies `204`
![image](https://github.com/harvester/tests/assets/2773781/dcdf80bc-40b0-4248-8a34-5e43ba9cef2d)

Reply:
the endpoint starts with `v1/harvester` is Harvester customized APIs, it will return `204` in DELETE operation, the usual endpoint in `apis` will return `200` with context like:
```
{'kind': 'Status', 'apiVersion': 'v1', 'metadata': {}, 'status': 'Success', 'details': {'name': 'u1', 'group': 'harvesterhci.io', 'kind': 'virtualmachinetemplates', 'uid': '5eda6fff-12ff-44e1-acab-632493a0fc25'}}
```

customized might cause unexpected behavior like https://github.com/harvester/harvester/issues/4641, so we would usually use endpoints in `apis/` (and as you can see, there would have version *v1beta1* in apis, but `v1/harvester` haven't).